grr/server/grr_response_server/gui/webauth.py

```diff
@@ -51,6 +53,57 @@ def RedirectBase(self):
     """Return a redirect to the main GRR page."""
     return werkzeug_utils.redirect(config.CONFIG["AdminUI.url"])
 
+class IAPWebAuthManager(BaseWebAuthManager):
+  """Auth Manager for Google IAP.
+
+  This extension pulls the x-goog-iap-jwt-assertion header and generates
+  a new user for that header via the 'sub' claim. Authorization is now 
+  delegated to IAP.
+  """
+
+  def __init__(self, *args, **kwargs):
+    super(IAPWebAuthManager, self).__init__(*args, **kwargs)
+
+    if config.CONFIG["AdminUI.cloud_project_id"] is None or \
+       config.CONFIG["AdminUI.cloud_backend_service_id"] is None:
+      raise RuntimeError("The necessary Cloud IAP configuration options are"
+                         "not set. Please set your AdminUI.cloud_project_id"
+                         "or AdminUI.cloud_backend_service_id keys.")
+
+    self.cloud_project_id = config.CONFIG["AdminUI.cloud_project_id"]
+    self.backend_service_id = config.CONFIG["AdminUI.cloud_backend_service_id"]
+    self.iap_header = "x-goog-iap-jwt-assertion"
+
+  def SecurityCheck(self, func, request, *args, **kwargs):
+    """Wrapping function."""
+    request.user = u""
```

Comment:
I don't think this is needed. If the user can't be authenticated, request.user can be simply left unset.

Reply:
👍

Reply:
Just a tip for the future: this module has `from __future__ import unicode_literals` import which means that unicode literals are enabled by default - the `u` prefix in this empty string is redundant.

Reply:
Makes sense to me!